### PR TITLE
feat: Force refresh personal access token

### PR DESCRIPTION
**Pull request by tolusha**
### What does this PR do?
feat: Force refresh personal access token

### What issues does this PR fix or reference?
https://github.com/eclipse-che/che/issues/23005

### Is it tested? How?

N/A

#### Release Notes

N/A

#### Docs PR

N/A

### CODE REVIEW

**Review by olexii4 — APPROVED** [review]

File: packages/dashboard-frontend/src/services/oauth/index.ts [hunk 1]
```diff
@@ -27,22 +27,16 @@ export class OAuthService {
...
-    }
-    if (!workspace.spec.template.projects) {
-      return;
-    }
-
-    const project = workspace.spec.template.projects[0];
-    if (!project || !project.git) {
-      return;
-    }
+  static async refreshTokenIfProjectExists(workspace: devfileApi.DevWorkspace): Promise<void> {
```
Comment (ibuziuk) [inline packages/dashboard-frontend/src/services/oauth/index.ts:30]: @tolusha I'm a bit lost, why project presence in devfile is indicator for token refresh ?

File: packages/dashboard-frontend/src/services/oauth/index.ts [hunk 2]
```diff
@@ -27,22 +27,16 @@ export class OAuthService {
...
+    // Find first git project.
+    let project = workspace.spec.template.projects?.find(project => !!project.git);
+    project = project || workspace.spec.template.starterProjects?.find(project => !!project.git);
+    project = project || workspace.spec.template.dependentProjects?.find(project => !!project.git);
 
     try {
-      await refreshFactoryOauthToken(project.git.remotes.origin);
+      if (project) {
+        await refreshFactoryOauthToken(project.git!.remotes.origin);
```
Comment (ibuziuk) [inline packages/dashboard-frontend/src/services/oauth/index.ts:38]: is it expected change?

File: packages/dashboard-frontend/src/services/oauth/index.ts (hunk 1 above)
Comment (ibuziuk) [reply packages/dashboard-frontend/src/services/oauth/index.ts:30]: at the very least, would be great to have some comment with method description

Comment (tolusha) [reply packages/dashboard-frontend/src/services/oauth/index.ts:30]: We need a repository url to figure out the correct OAuth provider.

File: packages/dashboard-frontend/src/services/oauth/index.ts (hunk 2 above)
Comment (tolusha) [reply packages/dashboard-frontend/src/services/oauth/index.ts:38]: Yes

**Review by ibuziuk — APPROVED** [review]

**Pull request merged**